### PR TITLE
Makespec: Fix version Number Processing

### PR DESCRIPTION
**Pull request by rieger07**
ran into a bug while using an alpha version of pycrypto (2.7a)
This code should fix it.
Also, found a typo when we log the exception

### CODE REVIEW

File: PyInstaller/building/makespec.py [hunk 1]
```diff
@@ -319,13 +319,11 @@ def main(scripts, name=None, onefile=None,
         # version is >= 2.4.
         try:
             import Crypto
-
-            pycrypto_version = list(map(int, Crypto.__version__.split('.')))
-            is_version_acceptable = pycrypto_version[0] >= 2 and pycrypto_version[1] >= 4
-
+            major, minor,_,_,_ = Crypto.version_info
+            temp_version = major*10+minor
+            is_version_acceptable = 24 <= temp_version
```
Comment (htgoebel) [inline PyInstaller/building/makespec.py]: Why not just
 `is_version_acceptable = (major > 2 and minor > 4)`
or
`is_version_acceptable = ((major, minor) > (2, 4))`
?

Comment (rieger07) [inline PyInstaller/building/makespec.py]: If we did it this way you could run into the problem of Pycrypto version 3.0 not passing.
3 >= 2 AND 0 >=4 evaluates to True AND False which evaluates to False

If you're willing to add distutils as a dependency you can use distutils.version.looseversion to worry about this kind of thing for you.

**Comment by rieger07** [comment]
If we did it this way you could run into the problem of Pycrypto version 3.0 not passing.
`3 >= 2 AND 0 >=4` evaluates to `True AND False` which evaluates to `False`

If you're willing to add distutils as a dependency you can use `distutils.version.LooseBersion` to worry about this kind of thing for you.
e.g.
`is_version_acceptable = LooseVersion('2.4') >= LooseVersion(Crypto.__version__)`

**Comment by htgoebel** [comment]
Oh, yes, me fool, of course the first example is wrong.

We are not using LooseVersion yet, but since distutils is a a standard package, I don't mind using it here.

**Comment by htgoebel** [comment]
@rieger07 Any progress?

**Comment by rieger07** [comment]
I updated to use looseversion, sorry for the delay.
Work got busy.

**Pull request closed without merging**